src/rpc/quorums.cpp

```diff
@@ -874,20 +875,37 @@ static UniValue verifychainlock(const JSONRPCRequest& request)
 
     const uint256 nBlockHash(ParseHashV(request.params[0], "blockHash"));
 
-    CBLSSignature sig;
-    if (!sig.SetHexStr(request.params[1].get_str())) {
-        throw JSONRPCError(RPC_INVALID_PARAMETER, "invalid signature format");
-    }
-
     int nBlockHeight;
+    CBlockIndex* pIndex{nullptr};
     if (request.params[2].isNull()) {
-        const CBlockIndex* pIndex = WITH_LOCK(cs_main, return g_chainman.m_blockman.LookupBlockIndex(nBlockHash));
+        pIndex = WITH_LOCK(cs_main, return g_chainman.m_blockman.LookupBlockIndex(nBlockHash));
         if (pIndex == nullptr) {
             throw JSONRPCError(RPC_INTERNAL_ERROR, "blockHash not found");
         }
         nBlockHeight = pIndex->nHeight;
     } else {
         nBlockHeight = ParseInt32V(request.params[2], "blockHeight");
+        LOCK(cs_main);
+        if (nBlockHeight < 0) {
+            throw JSONRPCError(RPC_INVALID_PARAMETER, "Block height out of range");
+        }
+        if (nBlockHeight <= ::ChainActive().Height()) {
+            pIndex = ::ChainActive()[nBlockHeight];
+        }
+    }
+
+    CBLSSignature sig;
+    if (pIndex) {
+        bool use_legacy_signature = !llmq::utils::IsV19Active(pIndex);
+        if (!sig.SetHexStr(request.params[1].get_str(), use_legacy_signature)) {
+            throw JSONRPCError(RPC_INVALID_PARAMETER, "invalid signature format");
+        }
+    } else {
+        if (!sig.SetHexStr(request.params[1].get_str(), false) &&
+                !sig.SetHexStr(request.params[1].get_str(), true)
+        ) {
+            throw JSONRPCError(RPC_INVALID_PARAMETER, "invalid signature format");
+        }
```

Comment:
why wasn't this adopted?

Reply:
because it is not equal code; this suggestion changes behaviour and cause for functional tests to fail. 


Check this commit:
```
    partial revert "refactor: simplification of rpc/quorums.cpp bcs pindex can't be nullptr"
```